### PR TITLE
Kingdom Overview screen, Towns/Castles Overview screen, wrong numbers displayed for heroes' primary skills

### PR DESCRIPTION
**Pull request by shprotru**
Fix for #2356

### CODE REVIEW

File: src/fheroes2/kingdom/kingdom_overview.cpp [hunk 1]
```diff
@@ -437,7 +437,9 @@ void StatsCastlesList::RedrawItem( const CstlRow & row, s32 dstx, s32 dsty, bool
 
         if ( hero ) {
             Interface::RedrawHeroesIcon( *hero, dstx + 82, dsty + 19 );
-            text.Set( hero->StringSkills( "-" ) );
+            char skillValues[64];
```
Comment (ihhub) [inline src/fheroes2/kingdom/kingdom_overview.cpp:440]: This is incorrect way of populating a string. Please do not use `sprintf` for such purposes but a normal string concatenation.

File: src/fheroes2/heroes/skill.cpp [hunk 2]
```diff
@@ -832,12 +825,12 @@ int Skill::GetLuckModifiers( int level, std::string * strs = NULL )
 
 StreamBase & Skill::operator<<( StreamBase & msg, const Primary & skill )
 {
-    return msg << skill.attack << skill.defense << skill.knowledge << skill.power;
```
Comment (ihhub) [inline src/fheroes2/heroes/skill.cpp:835]: Do we need this change and the change below? Aren't they below to file saving and loading?

File: src/fheroes2/heroes/skill.cpp [hunk 3]
```diff
@@ -220,13 +220,6 @@ std::string Skill::Primary::StringDescription( int skill, const Heroes * hero )
     return res;
 }
 
-std::string Skill::Primary::StringSkills( const std::string & sep ) const
-{
-    std::ostringstream os;
-    os << GetString( attack ) << sep << GetString( defense ) << sep << GetString( knowledge ) << sep << GetString( power );
-    return os.str();
```
Comment (ihhub) [inline src/fheroes2/heroes/skill.cpp:227]: return `GetAttack() + sep + GetDefense() + sep + GetPower() + sep + GetKnowledge();` will work as expected.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @shprotru I left few comments. Please take a look into them.

File: src/fheroes2/kingdom/kingdom_overview.cpp (hunk 1 above)
Comment (shprotru) [reply src/fheroes2/kingdom/kingdom_overview.cpp]: done

File: src/fheroes2/heroes/skill.cpp (hunk 2 above)
Comment (shprotru) [reply src/fheroes2/heroes/skill.cpp]: done

File: src/fheroes2/heroes/skill.cpp (hunk 3 above)
Comment (shprotru) [reply src/fheroes2/heroes/skill.cpp:227]: done

File: src/fheroes2/kingdom/kingdom_overview.cpp [hunk 4]
```diff
@@ -37,6 +37,11 @@
 #include "ui_button.h"
 #include "world.h"
 
+namespace
+{
+    const std::string sep = "-";
```
Comment (ihhub) [inline src/fheroes2/kingdom/kingdom_overview.cpp:42]: Just declare it locally before `text.Set`. This variable should use stack memory so making it static makes no sense.

Comment (shprotru) [reply src/fheroes2/kingdom/kingdom_overview.cpp]: done

**Comment by ihhub** [comment]
@shprotru thank you very much for fixing this issue!

**Pull request merged**